### PR TITLE
Alarm tweaks (history and statuses separated)

### PR DESCRIPTION
**Pull request by Jalle19**
@tomrosenback

### CODE REVIEW

File: app/modbus.mjs [hunk 1]
```diff
@@ -252,38 +252,46 @@ export const getDeviceInformation = async (modbusClient) => {
...
-                    alarms[code] = Object.assign({}, alarm)
-                }
-            } else {
-                alarms.push(alarm)
+        const result = await mutex.runExclusive(async () => modbusClient.readHoldingRegisters(register, alarmOffset))
+        const code = result.data[0]
+        const state = result.data[1]
+
+        // Skip unset alarm slots and unknown alarm types
+        if (AVAILABLE_ALARMS[code] === undefined) {
```
Comment (Jalle19) [inline app/modbus.mjs:268]: I dropped `code === 0` from here

File: app/modbus.mjs [hunk 2]
```diff
@@ -252,38 +252,46 @@ export const getDeviceInformation = async (modbusClient) => {
...
+            continue
+        }
+
+        let alarm = Object.assign({}, AVAILABLE_ALARMS[code])
+        alarm.state = state
+        alarm.date = new Date(`${result.data[2] + 2000}-${result.data[3]}-${result.data[4]} ${result.data[5]}:${result.data[6]}:00`)
```
Comment (Jalle19) [inline app/modbus.mjs:274]: I'll add a test for this. I still haven't verified that the timestamp is correct with regards to timezones, I'll do that as well. (assuming the control panel shows the local time :thinking: )

File: app/handlers.mjs [hunk 3]
```diff
@@ -19,7 +19,7 @@ export const summary = async (modbusClient, req, res) => {
         'readings': await getReadings(modbusClient),
         'settings': await getSettings(modbusClient),
         'deviceInformation': await getDeviceInformation(modbusClient),
-        'alarms': await getAlarmStatuses(modbusClient, true, false),
+        'alarmHistory': await getAlarmHistory(modbusClient, true, false),
```
Comment (tomrosenback) [inline app/handlers.mjs:22]: As you removed the need for onlyActive and distinct, the true, false should be removed from here.

File: app/mqtt.mjs [hunk 4]
```diff
@@ -46,14 +46,14 @@ export const publishValues = async (modbusClient, mqttClient) => {
...
-    const alarmStatuses = await getAlarmStatuses(modbusClient, false, true)
+    const alarmStatuses = await getAlarmStatuses(modbusClient)
 
     for (const [index, alarm] of Object.entries(alarmStatuses)) {
         const topicName = `${TOPIC_PREFIX_ALARM}/${alarm.name}`
 
         // Boolean values are changed to "ON" and "OFF" respectively since those are the
         // defaults for MQTT binary sensors in Home Assistant
-        topicMap[topicName] = alarm.state == 2 ? 'ON' : 'OFF'
+        topicMap[topicName] = alarm.state === 2 ? 'ON' : 'OFF'
```
Comment (Jalle19) [inline app/mqtt.mjs:56]: Turns out booleans and numbers are always returned as such, so we can always use `===` when comparing against Modbus results

**Review by tomrosenback — CHANGES_REQUESTED** [review]
Looks good except the function call to getAlarmHistory in relation to the definition

File: app/handlers.mjs (hunk 3 above)
Comment (Jalle19) [reply app/handlers.mjs:22]: Good catch!

File: app/modbus.mjs (hunk 2 above)
Comment (Jalle19) [reply app/modbus.mjs]: Added a test

File: app/handlers.mjs (hunk 3 above)
Comment (Jalle19) [reply app/handlers.mjs]: Fixed

**Review by tomrosenback — APPROVED** [review]
LGTM!

**Comment by Jalle19** [comment]
Yeah, timezone issues :D I'll sort it out

**Comment by Jalle19** [comment]
Tests should pass now

File: app/modbus.mjs [hunk 5]
```diff
@@ -30,25 +30,25 @@ const AVAILABLE_SETTINGS = {
...
+    10: {name: 'LTOFault', description: 'Heatexchanger fault'}, // 10=LTO vika
+    11: {name: 'CoolingFault', description: 'Cooling fault'}, // 11=Jäähdytys vika
+    12: {name: 'EmergencyStop', description: 'Emergency stop'}, // 12=Hätäseis
+    13: {name: 'FireRisk', description: 'Fire risk'}, // 13=Palovaara
+    14: {name: 'ServiceReminder', description: 'Service reminder'}, // 14=Huoltomuistutus
+    15: {name: 'HeaterPressureSwitch', description: 'Heater pressure switch'}, // 15=SLP painevahti
+    16: {name: 'InletFilterDirty', description: 'Inlet filter dirty'}, // 16=Tulosuodatin likainen,
+    17: {name: 'OutletFilterDirty', description: 'Outlet filter dirty'}, // 17=Poistosuodatin likainen
+    20: {name: 'InletFanPressureAbnomaly', description: 'Inlet fan pressure abnomaly'}, // 20=Tulopuhallin painepoikkeama
+    21: {name: 'OutletFanPressureAbnomaly', description: 'Outlet fan pressure abnomaly'}, // 21=Poistopuhallin painepoikkeama
```
Comment (Jalle19) [inline app/modbus.mjs:51]: These changes are by accident, I'll fix

**Review by tomrosenback — APPROVED** [review]

**Pull request merged**